rockcraft/extensions/gunicorn.py

```diff
@@ -125,6 +112,30 @@ def _gen_parts(self) -> dict:
             }
         return parts
 
+    def _check_async(self) -> str:
+        """Check if gevent package installed in requirements.txt.
+
+        Regex will match with exact `gevent` package and also
+        will match `gevent` with versions such as:
+          - gevent==*.*
+          - gevent ==0.*
+          - gevent <2.*
+          - gevent<2.*
+          - gevent >2.*
+          - gevent>2.*
+          - gevent@github.com/*
+          - gevent[some_extra]
+          - gevent [some_extra]
+        """
+        requirements_file = self.project_root / "requirements.txt"
+        with requirements_file.open() as f:
+            lines = f.readlines()
+            for line in lines:
+                for token in line.split():
+                    if re.match("^gevent[|[=|@|\\s|\n<|>]|(^gevent$)", token):
+                        return "gevent"
+        return "sync"
```

Comment:
I would propose that the algorithm here is:
1. Read the `requirements.txt` file using pathlib.Path(...).read_text()
2. Convert each line to a package name using the regex here: https://packaging.python.org/en/latest/specifications/name-normalization/#name-format
3. Normalize the names using https://packaging.python.org/en/latest/specifications/name-normalization/#name-format
4. Check if gevent is in the collection

Reply:
That is a great way! Thank you

Reply:
The regex here doesn't consider when versioning is used with package names. The regex I crafted makes sure user can use versioning or local/vcs with gevent package. https://packaging.python.org/en/latest/specifications/name-normalization/#name-format

I tested the regex with rockcrafts  own `requirements.txt` file but it was not able to find even 1 package since all of them are version pinned. [test code](https://pastebin.canonical.com/p/s4BpVkcXBT/)

Reply:
You can remove the `$` at the end of the regex and it should match the package name up to the version part